app/src/main/java/com/materiapps/gloom/domain/models/ModelStatus.kt

```diff
@@ -0,0 +1,29 @@
+package com.materiapps.gloom.domain.models
+
+import com.materiapps.gloom.ProfileQuery
+import com.materiapps.gloom.UserProfileQuery
+
+data class ModelStatus(
+    val emoji: String?,
+    val message: String?
+) {
+
+    companion object {
+
+        fun fromProfileQuery(pq: ProfileQuery.Data) = with(pq.viewer.status) {
+            if(this == null)
+                ModelStatus(null, null)
+            else
+                ModelStatus(emoji, message)
+        }
+
+        fun fromUserProfileQuery(upq: UserProfileQuery.Data) = with(upq.user?.status) {
+            if(this == null)
+                ModelStatus(null, null)
+            else
+                ModelStatus(emoji, message)
+        }
```

Comment:
guh formatting + curly braced ifs

Reply:
no curly braces needed, and formatting is fine

Reply:
IMO curly braces fit more into compose formatting

Reply:
i feel in this case its just unnecessary

Reply:
ugly